src/authentication/Credentials.ts

```diff
@@ -1,19 +1,8 @@
 /**
  * Credentials identifying an entity accessing or owning data.
  */
-export interface Credential {
-  webId?: string;
-}
-
-/**
- * Specific groups that can have credentials.
- */
-export enum CredentialGroup {
-  public = 'public',
-  agent = 'agent',
-}
-
-/**
- * A combination of multiple credentials, where their group is specified by the key.
- */
-export type CredentialSet = Partial<Record<CredentialGroup, Credential>>;
+export type Credentials = {
+  agent?: { webId: string };
+  client?: { webId: string };
```

Comment:
```suggestion
  client?: {  clientId: string };
```

This is a little confusing because I don’t think that Client ID documents are WebIDs.

They’re not usually RDF respurces for example.